### DISCUSSION TITLE
fetch post error

### DISCUSSION THREAD

**Discussion by wcly** [tauri-apps/tauri#5178] [Q&A] [answered]
my code like this:
```typescript
  try {
    const res = await fetch(url, {
       method: "POST",
       body: Body.json({
         item: 'xxx',
       }),
    });
    return res.data;
  } catch (err) {
    console.log(err);
  }
```

it course an error: 
```
invalid type: map, expected a string
```

what does that mean? how can I fix it?

**Comment by abisar** [comment, marked as answer]
Is that error from rust? It would be better if you posted the whole function and how you're feeding `res.data` from js to rust.
`res.data` is an object (or map in rust). My guess is, the rust side of your app doesn't like what you're giving it and wants a string and not the whole `res.data`.

**Reply by wcly** [reply]
whole function is:
```js
import { fetch, FetchOptions } from "@tauri-apps/api/http";

const request = async (url: string, options: FetchOptions): Promise<any> => {
  try {
    const env = localStorage.getItem(LOCAL_STORAGE_KEYS.ENV) ?? ENV.DEV;
    const baseURL = baseURLs[env as ENV].mobile;

    let headers = options.headers ?? {};

    if (!url.includes("public")) {
      const token = getAccessToken();
      headers = { ...headers, Authorization: token };
    }

    const res = await fetch(baseURL + url, {
      ...options,
      headers,
    });
    console.log('1111');
    return res.data;
  } catch (err: any) {
    throw Error(err);
  }
};

function getDevice(productKey: string): Promise<any> {
  return request("/device/get", {
    method: "POST",
    body: Body.json({
      productKey,
    }),
  });
}
```

then I just call funcion `getDevice`:
```js
const handleClick = async () => {
   await getDevice('xxx');
}
```

I think is not the return `res.data` trigger the error, casue `console.log('1111')` didn't trigger, it's the error from `fetch`. 
Don't not how to fix this.

**Reply by abisar** [reply]
I'm lost too. Give the other `Body` methods a try and see if you still get that error, maybe like this one:
`body: Body.text('xxx');`

**Reply by FabianLars** [reply]
Can you post the output of `tauri info` please? The script in your initial post is correct and shouldn't cause this issue, so if you see the same error with that small snippet too, something else must be wrong.

Also you may inspect your options & headers variable and make sure it doesn't contain maps/objects where it excepts strings. I still suspect a version mismatch of some sort tho 🤔

**Reply by wcly** [reply]
@FabianLars fixed, thanks, is my `getAccessToken` function, it's a `async` function, I forgot add `await` before it.
